File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long num = Math.abs((long) x);
        if(num>=2147483648L) return 0;
        long r=0;
        while(num!=0){
            long d= num%10;
            r=r*10+d;
            num=num/10;
        }
        if(r>=2147483648L) return 0;
        return x<=0? (int)(-1*r):(int) r;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        n=abs(x-0)
        r=0
        while(n!=0):
            d=n%10
            r=r*10+d
            n=n//10
        if(r>=2147483648):
            return 0
        if(x<0) :
            return -1*r
        return r%2147483648
        